driver/health/event_manager_default.go

```diff
@@ -0,0 +1,63 @@
+package health
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/ory/x/healthx"
+	"github.com/pkg/errors"
+)
+
+type DefaultHealthEventManager struct {
+	evtChan                chan interface{}
```

Comment:
By making this an interface with `HealthListenerID() string` or optionally a global enum we can avoid all of the reflection. This would be more along the lines of Go's "idiomatic" design. The problem with reflection is one of prefixing (I have encountered extremely hard to debug issues in gobufallo/pop related to this) and using an interface here is much more obvious to the (code) observer. What do you think?

Reply:
By type name clashes do you mean something like that:
```
./root/
 |-> a/
 |   `-> test/test.go      // defines a struct named Test
 `-> b/
     `-> test/test.go      // defines a struct also named Test
```
Meaning their types would both be `test.Test`

If that's what you are refering, reflect type comparaison is safe for that, I've joined a small demonstration:
![Capture d’écran de 2021-03-24 10-44-10](https://user-images.githubusercontent.com/27413532/112289391-90f2ad80-8c8e-11eb-8fe6-aff5e6cc2cfc.png)

However your idea of using an event interface is indeed quite clearer and will avoid the obfuscation caused by reflection, I'll update the code.

Reply:
Ah, that makes sense - I thought we're using string representation where this is a problem :)

> However your idea of using an event interface is indeed quite clearer and will avoid the obfuscation caused by reflection, I'll update the code.

Cool! :)

Reply:
I've done the changes :wink: